test/Tester/ClientConnectionTests/InvalidPreambleConnectionTests.cs

```diff
@@ -8,9 +12,10 @@ namespace Tester.ClientConnectionTests
     public class InvalidPreambleConnectionTests : TestClusterPerTest
     {
         [Fact, TestCategory("Functional")]
-        public void ShouldCloseConnectionWhenClientSendsInvalidPreambleSize()
+        public async Task ShouldCloseConnectionWhenClientSendsInvalidPreambleSize()
         {
-            var gwEndpoint = this.HostedCluster.Primary.NodeConfiguration.ProxyGatewayEndpoint;
+            var clusterConfig = await this.Client.GetGrain<ITestClusterManagementGrain>(Guid.Empty).GetService<ClusterConfiguration>();
+            var gwEndpoint = clusterConfig.Overrides[Silo.PrimarySiloName].ProxyGatewayEndpoint;
```

Comment:
There are other ways to get the proxy endpoint without the need to resort to serializing the config back from the silo (and in fact that would break if you stop using the legacy config).
If you hold on to the TestClusterOptions, you can peek in there. Otherwise there's also the opportunity to get the gateway list provider options from the client itself without asking the silo for the config